modin/engines/base/frame/partition_manager.py

```diff
@@ -555,6 +555,14 @@ def to_pandas(cls, partitions):
         else:
             return cls.concatenate(df_rows)
 
+    @classmethod
+    def wait_computations(cls, partitions):
```

Comment:
Along with changes above, I would move this outside the class as a decorator and decorate all functions, first checking if `BenchmarkMode.get()` then executing the line below. This way, it's not an API that is explicitly available for users.

Reply:
@devin-petersohn thanks for review! It's make sense. 

`wait_computations` was made as decorator function and other wait ways were removed. Please review again :)